src/MyInfoGovClient.class.ts

```diff
@@ -1,609 +1,348 @@
-    const basestring = this._formulateBaseString({
-      httpMethod: 'GET',
-      url,
-      appId: this.appId,
-      clientId: this.clientId,
-      singpassEserviceId,
-      nonce,
-      requestedAttributes,
-      timestamp,
-      txnNo,
-    })
+  /**
+   * Retrieves the given MyInfo attributes from the Person endpoint after
+   * the client has logged in to SingPass and consented to providing the given
+   * attributes.
+   * @param authCode Authorisation code given by MyInfo
+   * @param requestedAttributes Attributes to request from Myinfo. Should correspond
+   * to the attributes provided when initiating SingPass login.
+   * @returns Object containing access token used to retrieve the data,
+   * the NRIC/FIN and the data
+   */
+  async getPerson(
+    authCode: string,
+    requestedAttributes: string[],
+  ): Promise<IPersonResponse> {
+    // Obtain access token
+    let accessToken: string
+    try {
+      accessToken = await this._getAccessToken(authCode)
+    } catch (err) {
+      this._rethrowGetPersonError(
+        err,
+        'An error occurred while retrieving the access token from MyInfo',
+      )
+    }
 
-    // Generate the request signature by signing the constructed basestring
-    // with private key
-    const signature = this._signBaseString(
-      basestring,
-      this.privateKey,
-      'base64',
-    )
+    // Extract NRIC
+    let uinFin: string
+    try {
+      uinFin = this._extractUinFin(accessToken)
+    } catch (err) {
+      this._rethrowGetPersonError(
+        err,
+        'An error occurred while decoding the access token from MyInfo',
+      )
+    }
 
-    // Construct the authentication header using the signature, nonce and
-    // timestamp
-    const authHeader = this._formulateAuthHeader({
-      realm: this.realm,
-      appId: this.appId,
-      nonce,
-      signature,
-      timestamp,
-    })
+    // Get Person data
+    let data: IPerson
+    try {
+      data = await this._sendPersonRequest(
+        accessToken,
+        requestedAttributes,
+        uinFin,
+      )
+    } catch (err) {
+      this._rethrowGetPersonError(
+        err,
+        'An error occurred while calling the Person API',
+      )
+    }
+    return { accessToken, uinFin, data }
+  }
 
-    // Construct the request headers
-    const headers = {
-      'Content-Type': 'application/json',
-      Authorization: authHeader,
-      Accept: 'application/json',
+  /**
+   * Helper function for getPerson, which type-checks and re-throws errors.
+   * @param err The error object from a catch block
+   * @param defaultMsg String which will be prefixed to the error's message if
+   * the error is an Error object, otherwise wrapped in an Error and re-thrown
+   */
+  _rethrowGetPersonError(err: any, defaultMsg: string): never {
+    if (err instanceof Error) {
+      err.message = `${defaultMsg}: ${err.message}`
+      throw err
     }
+    throw new Error(`${defaultMsg}: ${JSON.stringify(err)}`)
+  }
```

Comment:
would it be better to introduce custom error types instead, like the main Forms codebase?